balancer-js/src/modules/zaps/bbausd2-migrations/bbausd1.integration.spec.ts

```diff
@@ -0,0 +1,314 @@
+  token: string,
+  from: string,
+  to: string
+): Promise<BigNumber> => {
+  const holder = await impersonateAccount(from);
+  const balance = await getErc20Balance(token, from);
+  await contracts.ERC20(token, provider).connect(holder).transfer(to, balance);
+
+  return balance;
+};
+
+const stake = async (
+  signer: JsonRpcSigner,
+  balance: BigNumber
+): Promise<void> => {
+  await (
+    await contracts
+      .ERC20(poolAddress, provider)
+      .connect(signer)
+      .approve(gaugeAddress, MaxUint256)
+  ).wait();
+
+  await (
+    await signer.sendTransaction({
+      to: gaugeAddress,
+      data: liquidityGauge.encodeFunctionData('deposit', [balance]),
+    })
+  ).wait();
+};
+
+describe('bbausd migration execution', async () => {
+  let signer: JsonRpcSigner;
+  let signerAddress: string;
+  let authorisation: string;
+  let balance: BigNumber;
+  let pool: PoolModel;
+
+  beforeEach(async function () {
+    this.timeout(20000);
+
+    await reset();
+
+    signer = provider.getSigner();
+    signerAddress = await signer.getAddress();
+    authorisation = await signRelayerApproval(relayer, signerAddress, signer);
+    // Transfer tokens from existing user account to signer
+    // We need that to test signatures, because hardhat doesn't have impersonated accounts private keys
+    balance = await move(poolAddress, holderAddress, signerAddress);
+
+    const config = {
+      network,
+      rpcUrl,
+    };
+    const subgraph = new Subgraph(config);
+    const pools = new PoolsProvider(
+      config,
+      new SubgraphPoolRepository(subgraph.client)
+    );
+    const _pool = await pools.findBy('address', poolAddress);
+    if (!_pool) throw new BalancerError(BalancerErrorCode.POOL_DOESNT_EXIST);
+    pool = _pool;
+  });
+
+  async function testFlow(
+    staked: boolean,
+    authorized = true,
+    minBbausd2Out: undefined | string = undefined
+  ): Promise<string> {
+    const addressIn = staked
+      ? addresses.bbausd1.gauge
+      : addresses.bbausd1.address;
+    const addressOut = staked
+      ? addresses.bbausd2.gauge
+      : addresses.bbausd2.address;
+    // Store balance before migration
+    const before = {
+      from: await getErc20Balance(addressIn, signerAddress),
+      to: await getErc20Balance(addressOut, signerAddress),
+    };
+
+    const amount = before.from;
+
+    let query = migrations.bbaUsd(
+      signerAddress,
+      amount.toString(),
+      '0',
+      staked,
+      pool.tokens
+        .filter((token) => token.symbol !== 'bb-a-USD') // Note that bbausd is removed
+        .map((token) => parseFixed(token.balance, token.decimals).toString()),
+      authorisation
+    );
+
+    const gasLimit = MAX_GAS_LIMIT;
+
+    // Static call can be used to simulate tx and get expected BPT in/out deltas
+    const staticResult = await signer.call({
+      to: query.to,
+      data: query.data,
+      gasLimit,
+    });
+    const expectedBpts = query.decode(staticResult, staked);
```

Comment:
can we cast expected result type?

Reply:
This expectedBpts is a temporary solution to test while bbausd1 and bbausd2 are the same pools. When the new pool gets created, we'll be able to refactor that decode function to return a string like the other flows. By then, I believe we should cast it to a string 👍